napari/layers/tracks/tracks.py

```diff
@@ -0,0 +1,489 @@
+# from napari.layers.base.base import Layer
+# from napari.utils.events import Event
+# from napari.utils.colormaps import AVAILABLE_COLORMAPS
+
+from typing import Union
+
+import numpy as np
+
+from ...utils.colormaps import AVAILABLE_COLORMAPS
+from ...utils.events import Event
+from ..base import Layer
+from ._track_utils import TrackManager, get_track_dimensionality
+
+ALLOW_ND_SLICING = False
+
+
+class Tracks(Layer):
+    """ Tracks
+
+    A Tracks layer for overlaying trajectories on image data.
+
+    Parameters
+    ----------
+    data : list
```

Comment:
How similar is this data representation to what was discussed here https://github.com/napari/napari/issues/539#issuecomment-564328904?

As I look in more I'm now wondering if my request about the properties `list` -> `dict` change doesn't make sense.

Maybe as we merge this you could make a contribution to the applications section of our tutorials repo https://github.com/napari/tutorials/tree/master/applications that explains how to use the tracks layer with tracks data from the [cell tracking dataset](https://public.celltrackingchallenge.net/documents/Naming%20and%20file%20content%20conventions.pdf) or an [SWC](http://www.neuronland.org/NLMorphologyConverter/MorphologyFormats/SWC/Spec.html) file?

Reply:
I think the `list`/`dict` question for the track `properties` is the biggest question in terms of the public interface.  I've tried to make it agnostic to the potential application. Much of the discussion in #539 was for cell tracking, with specialist terms like 'lineage' - these concepts are now encapsulated in a more general `graph` which specifies edges between the track nodes via the `parent` property.

Having an applications tutorial sounds like a great idea!

Reply:
Another idea is that the `properties` and `graph` could be separate kwargs for the layer e.g.:

```python
add_tracks(data, properties=properties, graph=graph)
```

Reply:
> Another idea is that the properties and graph could be separate kwargs for the layer e.g.:

I think it fine to keep with just `properties` for now, if we can get it working with the dataframe like representation, but if not then splitting out may make sense. It's always easier to split out later if we want